src/cos_integration.py

```diff
@@ -102,43 +102,33 @@ def get_metrics_endpoints(self, node_name: str, token: str, cluster_name: str) -
 
         kubernetes_jobs = [
             JobConfig(
-                "kube-proxy",
-                "/metrics",
-                "http",
-                "localhost:10249",
-                [{"target_label": "job", "replacement": "kube-proxy"}],
+                name="apiserver",
+                metrics_path="/metrics",
+                scheme="https",
+                target="localhost:6443",
+                relabel_configs=[{"target_label": "job", "replacement": "apiserver"}, instance_relabel],
             ),
             JobConfig(
-                "apiserver",
-                "/metrics",
-                "https",
-                "localhost:6443",
-                [
-                    {
-                        "source_labels": ["job"],
```

Comment:
did we lose `source_labels` for the `apiserver`?

Reply:
This is not required. In the case of adding source_labels, Prometheus requires that a `job` label already exists in the original metrics to replace it. However, if we omit the `source_labels`, Prometheus will unconditionally set `job="apiserver"`, regardless of whether the label existed before.

Reply:
it's wild how much this charm needs to know about prometheus in order to ship metrics to it.  But it seems important.  Thanks for the details.